src/domain/chains/chains.repository.ts

```diff
@@ -16,15 +16,29 @@ import {
 } from '@/domain/indexing/entities/indexing-status.entity';
 import { ILoggingService, LoggingService } from '@/logging/logging.interface';
 import { differenceBy } from 'lodash';
+import { PaginationData } from '@/routes/common/pagination/pagination.data';
+import { IConfigurationService } from '@/config/configuration.service.interface';
 
 @Injectable()
 export class ChainsRepository implements IChainsRepository {
+  private readonly maxLimit: number;
```

Comment:
Nit: Why is this named `max`? I mean, it seems the default limit we use for all the requests to get the chains. Maybe I'm wrong but I'd expect this to be something like a threshold but it is like a constant. Is there any reason to not use `PaginationData.DEFAULT_LIMIT` for this?

Reply:
It mirrors the [`max_limit` of the `Chains` pagination from the Config. Service](https://github.com/safe-global/safe-config-service/blob/main/src/chains/views.py#L16).

I did consider changing the `PaginationData.DEFAULT_LIMIT` instead but that affects all pagination requests across the project. If we did, it would reduce the request by the Web though as they don't provide a `limit` in their request.

What do you think?

Reply:
I meant doing something like:
```typescript
const result = await this.getChains(PaginationData.DEFAULT_LIMIT, offset);
```

I didn't mean to change the `PaginationData.DEFAULT_LIMIT`, that would indeed affect the overall service behavior. Sorry about the lack of clarity in my previous comment 😄

Reply:
The use of `40` was [at the request of @PooyaRaki](https://github.com/safe-global/safe-client-gateway/pull/2002#discussion_r1795022413).

@PooyaRaki, what do you think about this?

Reply:
Changing the default limit could be risky, as it would impact all endpoints. However, since the web isn't specifying any limits, for this particular case, I think it's better to set a maximum limit of for example 40 specifically for this endpoint.

Reply:
Considering the above, I've moved the limit to a constant within the repository as I don't see it changing (for now). I left the sequential page requests in the configuration though.

What are your thoughts?

Reply:
Sounds good to me! 🚀